### PR TITLE
Refactor NPC debug menu, better targeting

### PR DESCRIPTION
**Pull request by Coolthulhu**
Moved the entire NPC debug menu (accessible with debug->d) to debug_menu.cpp.

Changed targeting: instead of having to manually find the NPC and "aim" at it, all available targets (player and NPCs) are listed in `uimenu`, similar to how chatting works.

### CODE REVIEW

File: src/debug_menu.cpp [hunk 1]
```diff
@@ -44,3 +50,333 @@ void debug_menu::teleport_overmap()
...
+
+            if( bp_ptr != nullptr ) {
+                int value;
+                if( query_int( value, "Set the stat to? Currently: %d", *bp_ptr ) && value >= 0 ) {
+                    *bp_ptr = value;
+                    p.reset_stats();
+                }
+            }
+        }
+        break;
```
Comment (BrettDong) [inline src/debug_menu.cpp]: Inconsistent indent of `break;` statement.

File: src/debug_menu.cpp [hunk 2]
```diff
@@ -44,3 +50,333 @@ void debug_menu::teleport_overmap()
...
+        case D_HP: {
+            uimenu smenu;
+            smenu.return_invalid = true;
+            smenu.addentry( 0, true, 'q', "%s: %d", _( "Torso" ), p.hp_cur[hp_torso] );
+            smenu.addentry( 1, true, 'w', "%s: %d", _( "Head" ), p.hp_cur[hp_head] );
+            smenu.addentry( 2, true, 'a', "%s: %d", _( "Left arm" ), p.hp_cur[hp_arm_l] );
+            smenu.addentry( 3, true, 's', "%s: %d", _( "Right arm" ), p.hp_cur[hp_arm_r] );
+            smenu.addentry( 4, true, 'z', "%s: %d", _( "Left leg" ), p.hp_cur[hp_leg_l] );
+            smenu.addentry( 5, true, 'x', "%s: %d", _( "Right leg" ), p.hp_cur[hp_leg_r] );
+            smenu.addentry( 999, true, 'q', "%s", _( "[q]uit" ) );
```
Comment (BrettDong) [inline src/debug_menu.cpp]: Conflicting letters: `q` is both assigned in "Torso"  and "[q]uit"

File: src/debug_menu.cpp [hunk 3]
```diff
@@ -44,3 +50,333 @@ void debug_menu::teleport_overmap()
...
+                    p.reset_stats();
+                }
+            }
+        }
+        break;
+        case D_ITEMS:
+            wishitem( &p );
+            break;
+        case D_DELETE_ITEMS:
+            if( !query_yn( "Delete all items from the target?" ) ) {
```
Comment (illi-kun) [inline src/debug_menu.cpp]: Non-translatable string, _() is required.

File: src/debug_menu.cpp [hunk 4]
```diff
@@ -44,3 +50,333 @@ void debug_menu::teleport_overmap()
...
+                case 3:
+                    bp_ptr = &p.per_max;
+                    break;
+                default:
+                    break;
+            }
+
+            if( bp_ptr != nullptr ) {
+                int value;
+                if( query_int( value, "Set the stat to? Currently: %d", *bp_ptr ) && value >= 0 ) {
```
Comment (illi-kun) [inline src/debug_menu.cpp]: Non-translatable string, _() is required.

File: src/debug_menu.cpp [hunk 5]
```diff
@@ -44,3 +50,333 @@ void debug_menu::teleport_overmap()
...
+                if( query_int( value, "Set the hitpoints to? Currently: %d", *bp_ptr ) && value >= 0 ) {
+                    *bp_ptr = value;
+                    p.reset_stats();
+                }
+            }
+        }
+        break;
+        case D_PAIN: {
+            int value;
+            if( query_int( value, "Cause how much pain? pain: %d", p.get_pain() ) ) {
```
Comment (illi-kun) [inline src/debug_menu.cpp]: Actually, all entries of ```query_int``` are missing _(), is it intended for some reason?

Comment (Coolthulhu) [inline src/debug_menu.cpp]: Probably not.

Comment (BrettDong) [inline src/debug_menu.cpp]: Sorry.

**Comment by Coolthulhu** [comment]
Continued in #19530

**Pull request closed without merging**